### PR TITLE
SG-27562 SG Desktop please add indication of app launch

### PR DESCRIPTION
**Pull request by NorberMV**
This PR introduces a frameless QDialog with an overlay embedded widget that is displayed temporarily when clicking on a DCC to launch it from SG Desktop indicating that the command used to start the DCC was successfully executed.
This is implemented for apps launched via tk-multi-launchapp.

Group 2
Rectangle 3

### CODE REVIEW

File: python/tk_multi_launchapp/base_launcher.py [hunk 1]
```diff
@@ -287,6 +292,33 @@ def _launch_app(
...
+            # Start spinner
+            wid.start_progress()
+            # Report progress
+            splash_message = "Launching executable '%s'" % (app_path)
+            wid.report_progress(0.00, splash_message)
+            splash_msg = "Launched successfully"
+            QtCore.QTimer.singleShot(
+                7000,
+                lambda: wid.report_progress(0.97, splash_msg),
+            )
```
Comment (carlos-villavicencio-adsk) [inline python/tk_multi_launchapp/base_launcher.py:318]: I see this successful message will always be displayed. However, what will happen for the use case of this problem https://jira.autodesk.com/browse/SG-30313 ?

File: python/tk_multi_launchapp/base_launcher.py [hunk 2]
```diff
@@ -287,6 +292,33 @@ def _launch_app(
...
+            # Hide the QDialog
+            QtCore.QTimer.singleShot(10000, lambda: dial.hide())
```
Comment (carlos-villavicencio-adsk) [inline python/tk_multi_launchapp/base_launcher.py:320]: I'm not sure how QT works here, but is the dialog being destroyed (freed memory) or just being hidden? Do you know if this has an impact on performance?

File: python/tk_multi_launchapp/base_launcher.py (hunk 1 above)
Comment (NorberMV) [reply python/tk_multi_launchapp/base_launcher.py:318]: The new launch indicator QDialog will only be displayed once the command executed by app_launch hook has succeeded. If this command fails for some reason another dialog called 'not_found_dialog' will be displayed with the associated command error. 
The condition that evaluates to this is on the line 239.

File: python/tk_multi_launchapp/base_launcher.py (hunk 2 above)
Comment (NorberMV) [reply python/tk_multi_launchapp/base_launcher.py]: Good catch, thanks for the suggestion !, I have changed it by this method reject() which looks like triggers a close event and delete the dialog while it is visible.

**Review by rob-aitchison — APPROVED** [review]
All looks good - thank you!

File: python/launch_indicator_dialog/launch_spinner.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    Launcher indicator UI dialog.
+    """
+
+    def __init__(self, app_instance=None):
+        """
+        Constructor
+        """
+
+        # first, call the base class and let it do its thing.
+        QtGui.QDialog.__init__(self)
```
[the remark this answers is not in the document]
Comment (NorberMV) [reply python/launch_indicator_dialog/launch_spinner.py:52]: When calling _create_dialog_with_widget() in line 25, this new QDialog is wrapped into its parent TankQDialog() from tk-core, so it's not needed initialize it here, just the base class.

**Pull request merged**